Script started on Mon 20 May 2019 12:36:04 PM EDT
[1;31mLmod Warning: [0m The environment MODULEPATH has been changed in unexpected ways.
Lmod is unable to use given MODULEPATH. It is using:

"/apps/modulefiles/standard/compiler/gcc/7.1.0:/nv/t1/modulefiles:/apps/modulefiles/standard/core:/apps/modulefiles/standard/toolchains:/apps/modulefiles/vendor".

Please use "module use ..." to change MODULEPATH instead. 




Due to MODULEPATH changes the following have been reloaded:
  1) bowtie2/2.2.9

To execute picard run: java -jar $EBROOTPICARD/picard.jar
    To execute trimmomatic run: java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.36.jar
    
    
Due to MODULEPATH changes the following have been reloaded:
  1) bowtie2/2.2.9

^X^C/usr/bin/lua: /usr/share/lmod/lmod/libexec/../tools/base64.lua:34: interrupted!
stack traceback:
	/usr/share/lmod/lmod/libexec/../tools/base64.lua:28: in function 'bit'
	/usr/share/lmod/lmod/libexec/../tools/base64.lua:34: in function 'lor'
	/usr/share/lmod/lmod/libexec/../tools/base64.lua:83: in function 'decode64'
	/usr/share/lmod/lmod/libexec/utils.lua:372: in function 'getMT'
	/usr/share/lmod/lmod/libexec/lmod:682: in function 'main'
	/usr/share/lmod/lmod/libexec/lmod:711: in main chunk
	[C]: ?
[?1034h[30m[[33m12:36 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mexitcat $PROCESSED/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold1/PEPATAC_log.md[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[Cexit[K[Kscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/narrowPeakToBigBed.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks /results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes bedToBigBed /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_pea ks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed[A[ARscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/narrowPeakToBigBed.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes bedToBigBed /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed[A[A


[30m[[33m12:36 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mls -lhtr
total 144M
-rw-rw-r-- 1 jps3dp shefflab 34M May 20 11:35 gold2_peaks.xls
-rw-rw-r-- 1 jps3dp shefflab 30M May 20 11:35 gold2_peaks.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab 20M May 20 11:35 gold2_summits.bed
-rw-rw-r-- 1 jps3dp shefflab 30M May 20 11:35 gold2_peaks_fixedWidth.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab 31M May 20 11:35 gold2_peaks_fixedWidth_normalized.narrowPeak
[30m[[33m12:36 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mpwd
/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38
[30m[[33m12:36 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mls -lhtr
total 144M
-rw-rw-r-- 1 jps3dp shefflab 34M May 20 11:35 gold2_peaks.xls
-rw-rw-r-- 1 jps3dp shefflab 30M May 20 11:35 gold2_peaks.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab 20M May 20 11:35 gold2_summits.bed
-rw-rw-r-- 1 jps3dp shefflab 30M May 20 11:35 gold2_peaks_fixedWidth.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab 31M May 20 11:35 gold2_peaks_fixedWidth_normalized.narrowPeak
[30m[[33m12:36 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mls -lhtr[5Ppwdls -lhtrRscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/narrowPeakToBigBed.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peakss/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes bedToBigBed /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed
[30m[[33m12:37 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mls -lht[K[K[K[K
gold2_peaks_fixedWidth.narrowPeak  gold2_peaks_fixedWidth_normalized.narrowPeak  gold2_peaks.narrowPeak  gold2_peaks.xls  gold2_summits.bed
[30m[[33m12:37 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mls Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/narrowPeakToBigBed.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peakss/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes bedToBigBed /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed
pass1 - making usageList (90 chroms): 65 millis
pass2 - checking and writing primary data (357682 records, 10 fields): 855 millis
[1] TRUE
[1] TRUE
[30m[[33m12:38 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mls -lhtr
total 148M
-rw-rw-r-- 1 jps3dp shefflab  34M May 20 11:35 gold2_peaks.xls
-rw-rw-r-- 1 jps3dp shefflab  30M May 20 11:35 gold2_peaks.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab  20M May 20 11:35 gold2_summits.bed
-rw-rw-r-- 1 jps3dp shefflab  30M May 20 11:35 gold2_peaks_fixedWidth.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab  31M May 20 11:35 gold2_peaks_fixedWidth_normalized.narrowPeak
-rw-rw-r-- 1 jps3dp shefflab 6.2M May 20 12:38 gold2_peaks.bigBed
[30m[[33m12:38 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mlooper run /scratch/jps3dp/DATA/gold_hg38.yaml 
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Traceback (most recent call last):
  File "/home/jps3dp/.local/bin/looper", line 11, in <module>
    sys.exit(main())
  File "/home/jps3dp/.local/lib/python2.7/site-packages/looper/looper.py", line 807, in main
    file_checks=args.file_checks, compute_env_file=getattr(args, 'env', None))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/looper/project.py", line 40, in __init__
    no_compute_exception=RuntimeError, **kwargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/peppy/project.py", line 233, in __init__
    self[self_table_attr] = self.parse_sample_sheet(path_anns_file)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/peppy/project.py", line 1039, in parse_sample_sheet
    raise Project.MissingSampleSheetError(sample_file)
peppy.project.MissingSampleSheetError: Missing sample annotation sheet (/home/jps3dp/code//pepatac_testing/metadata/gold_atac_annotation.csv); a project need not use a sample sheet, but if it does the file must exist.
[0m[30m[[33m12:43 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mpeak_calling_hg38(B[m[30m](B[m$ (B[mcd ../
[30m[[33m12:43 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mecho $CODE
/home/jps3dp/code/
[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msource ~/,.[K[K.bashrc
[1;31mLmod Warning: [0m The environment MODULEPATH has been changed in unexpected ways.
Lmod is unable to use given MODULEPATH. It is using:

"/apps/modulefiles/standard/compiler/gcc/7.1.0:/nv/t1/modulefiles:/apps/modulefiles/standard/core:/apps/modulefiles/standard/toolchains:/apps/modulefiles/vendor".

Please use "module use ..." to change MODULEPATH instead. 




Due to MODULEPATH changes the following have been reloaded:
  1) bowtie2/2.2.9

To execute picard run: java -jar $EBROOTPICARD/picard.jar
    To execute trimmomatic run: java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.36.jar
    
    
Due to MODULEPATH changes the following have been reloaded:
  1) bowtie2/2.2.9

[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msource ~/.bashrc[6Pecho $CODE[4Pcd ../looper run /scratch/jps3dp/DATA/gold_hg38.yaml 
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Job script (n=1; 1.57 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711284
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Skipping sample 'gold5' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
4 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 1 of 5
Jobs submitted: 1
[0m[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md 
    pm.run(cmd, bigNarrowPeak, nofail=False, container=pm.container)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 774, in run
    self.callprint(cmd, shell, lock_file, nofail, container)  # Run command
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1011, in callprint
    self._triage_error(SubprocessError(msg), nofail)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 2069, in _triage_error
    self.fail_pipeline(e)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
pypiper.exceptions.SubprocessError: Subprocess returned nonzero result. Check above output for details
^C
[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4711284  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710495  standard pepatac.   jps3dp  R    2:04:04      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:04:04      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:04:34      1 udc-ba25-12a
           4710494  standard pepatac.   jps3dp  R    2:04:34      1 udc-ba27-32a
[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dptail -f PEPATAC_log.md 
    pm.run(cmd, bigNarrowPeak, nofail=False, container=pm.container)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 774, in run
    self.callprint(cmd, shell, lock_file, nofail, container)  # Run command
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1011, in callprint
    self._triage_error(SubprocessError(msg), nofail)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 2069, in _triage_error
    self.fail_pipeline(e)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
pypiper.exceptions.SubprocessError: Subprocess returned nonzero result. Check above output for details
### [Pipeline run code and environment:]

*              Command:  `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.py --single-or-paired paired --genome hg38 --sample-name gold2 --input /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz --input2 /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz --prealignments rCRSd_3k human_repeats --peak-type fixed --genome-size hs -O /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/results_pipeline -P 8 -M 16000`
*         Compute host:  udc-aj40-4c0
*          Working dir:  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2
*            Outfolder:  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/
*  Pipeline started at:   (05-20 12:44:38) elapsed: 1.0 _TIME_

### [Version log:]

*       Python version:  2.7.15
*          Pypiper dir:  `/sfs/qumulo/qhome/jps3dp/.local/lib/python2.7/site-packages/pypiper`
*      Pypiper version:  0.11.0
*         Pipeline dir:  `/sfs/lustre/scratch/jps3dp/tools/databio/pepatac/pipelines`
*     Pipeline version:  0.8.6
*        Pipeline hash:  a147ae41573786561ef0bfc35518da8210175b17
*      Pipeline branch:  * dev
*        Pipeline date:  2019-05-16 15:52:56 -0400
*        Pipeline diff:  3 files changed, 198 insertions(+), 167 deletions(-)

### [Arguments passed to pipeline:]

*             `input2`:  `['/sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz']`
*         `paired_end`:  `True`
*              `motif`:  `False`
*               `keep`:  `False`
*      `prealignments`:  `['rCRSd_3k', 'human_repeats']`
*          `peak_type`:  `fixed`
*          `new_start`:  `False`
*            `recover`:  `False`
*     `frip_ref_peaks`:  `None`
*           `TSS_name`:  `None`
*       `deduplicator`:  `samblaster`
*        `sample_name`:  `gold2`
*               `lite`:  `False`
*              `input`:  `['/sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz']`
*        `genome_size`:  `hs`
*        `config_file`:  `pepatac.yaml`
*             `extend`:  `250`
*                `mem`:  `16000`
*       `force_follow`:  `False`
*          `blacklist`:  `None`
*              `cores`:  `8`
*          `anno_name`:  `None`
*            `trimmer`:  `skewer`
*        `peak_caller`:  `macs2`
*      `output_parent`:  `/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/results_pipeline`
*   `single_or_paired`:  `paired`
*              `dirty`:  `False`
*    `genome_assembly`:  `hg38`
*            `no_fifo`:  `False`

----------------------------------------


Changed status from initializing to running.

Loading config file: /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.yaml

Local input file: /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz
Local input file: /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz

> `File_mb`	1603.8181	PEPATAC	_RES_

> `Read_type`	paired	PEPATAC	_RES_

> `Genome`	hg38	PEPATAC	_RES_

### Merge/link and fastq conversion:  (05-20 12:44:38) elapsed: 1.0 _TIME_

Number of input file sets: 2
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz`
Command ID incremented by: `1`. Current ID: `1`

Local input file: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz'
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `2`

Local input file: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz'
Found .fastq.gz file
Found .fq.gz file; no conversion necessary
Found .fastq.gz file
Found .fq.gz file; no conversion necessary
['ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatanged status from initializing to running.

Loading config file: /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.yaml

Local input file: /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz
Local input file: /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz

> `File_mb`	1603.8181	PEPATAC	_RES_

> `Read_type`	paired	PEPATAC	_RES_

> `Genome`	hg38	PEPATAC	_RES_

### Merge/link and fastq conversion:  (05-20 12:44:38) elapsed: 1.0 _TIME_

Number of input file sets: 2
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz`
Command ID incremented by: `1`. Current ID: `1`

Local input file: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz'
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `2`

Local input file: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz'
Found .fastq.gz file
Found .fq.gz file; no conversion necessary
Found .fastq.gz file
Found .fq.gz file; no conversion necessary
['ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz', 'ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz']
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz`
Command ID incremented by: `1`. Current ID: `3`

Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `4`


### Adapter trimming:  (05-20 12:44:38) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq`
Skipped command: `skewer -f sanger -t 8 -m pe -x /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/NexteraPE-PE.fa --quiet -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `5`

Skipped command: `mv /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2-trimmed-pair1.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq`
Command ID incremented by: `1`. Current ID: `6`

Skipped command: `mv /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2-trimmed-pair2.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2_trim.fastq`
Command ID incremented by: `1`. Current ID: `7`


### Prealignments (05-20 12:44:38) elapsed: 0.0 _TIME_

Prealignment assemblies: ['rCRSd_3k', 'human_repeats']

### Map to rCRSd_3k (05-20 12:44:38) elapsed: 0.0 _TIME_

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2`


> `mkfifo /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2` (266630)

<pre>
</pre>
Command completed. Elapsed time: 0:00:00. Running peak memory: 0.0GB.
  PID: 266630;	Command: mkfifo;	Return code: 0;	Memory used: 0.0GB
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq.gz`
Skipped command: `perl /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2_trim.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `9`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq.gz`
Skipped command: `(bowtie2 -p 8 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nm/t1/genomes/rCRSd_3k/indexed_bowtie2/rCRSd_3k --rg-id gold2 -U /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq --un /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log`
Command ID incremented by: `1`. Current ID: `10`


> `grep 'aligned exactly 1 time' /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log | awk '{print $1}'`


> `Aligned_reads_rCRSd_3k`	11074130.0	PEPATAC	_RES_

> `Alignment_rate_rCRSd_3k`	38.98	PEPATAC	_RES_

### Map to human_repeats (05-20 12:44:39) elapsed: 0.0 _TIME_

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2`


> `mkfifo /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2` (266634)

<pre>
</pre>
Command completed. Elapsed time: 0:00:ac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq --un /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_bt_aln_summary.log`
Command ID incremented by: `1`. Current ID: `13`


> `grep 'aligned exactly 1 time' /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_bt_aln_summary.log | awk '{print $1}'`


> `Aligned_reads_human_repeats`	1147860.0	PEPATAC	_RES_

> `Alignment_rate_human_repeats`	4.04	PEPATAC	_RES_

### Compress all unmapped read files (05-20 12:44:39) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq`
Command ID incremented by: `1`. Current ID: `14`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `15`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq`
Command ID incremented by: `1`. Current ID: `16`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `17`


### Map to genome (05-20 12:44:39) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam`
Skipped command: `bowtie2 -p 8  --very-sensitive -X 2000 --rg-id gold2 -x /nm/t1/genomes/hg38/indexed_bowtie2/hg38 -1 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq.gz -2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq.gz | samtools view -bS - -@ 1  | samtools sort - -@ 1 -T /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpMAVQYS -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam`
Command ID incremented by: `3`. Current ID: `20`

Skipped command: `samtools view -b -q 10 -@ 8 -U /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_fail_qc.bam -f 2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam`
Command ID incremented by: `1`. Current ID: `21`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam.bai`
Skipped command: `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam`
Command ID incremented by: `1`. Current ID: `22`

Skipped command: `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam`
Command ID incremented by: `1`. Current ID: `23`


> `samtools idxstats /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam | grep -we 'chrM' -we 'chrMT' -we 'M' -we 'MT' -we 'rCRSd'| cut -f 3`


> `Mitochondrial_reads`	801.0	PEPATAC	_RES_
Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam`


> `samtools idxstats /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam | cut -f 1 | grep -vwe 'chrM' -vwe 'chrMT' -vwe 'M' -vwe 'MT' -vwe 'rCRSd'| xargs samtools view -b -@ 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam` (266676,266677,266678,266679)

<pre>
^C
[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp  R    2:04:35      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:04:35      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:05:05      1 udc-ba25-12a
           4710494  standard pepatac.   jps3dp  R    2:05:05      1 udc-ba27-32a
           4711284  standard pepatac.   jps3dp  R       0:19      1 udc-aj40-4c0
[30m[[33m12:44 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msacct --format="JOBID, JOnb[K[K[K[KN[KJOBNAME%30"
       JobID                        JobName 
------------ ------------------------------ 
2870870              pepatac.py_melanoma_22 
2870889      pepatac.py_pancreatic_cancer_1 
2870942       pepatac.py_synovial_sarcoma_8 
2870943       pepatac.py_synovial_sarcoma_9 
2871353                  pepatac.py_AML_433 
2871354                  pepatac.py_AML_447 
2871355                  pepatac.py_AML_452 
2871356                  pepatac.py_AML_453 
2871357                  pepatac.py_AML_454 
2871358                  pepatac.py_AML_461 
2871359                    pepatac.py_ALL_1 
2871360                    pepatac.py_ALL_2 
2871361                    pepatac.py_ALL_3 
2871362                    pepatac.py_ALL_4 
2871363                    pepatac.py_ALL_5 
2871364                    pepatac.py_ALL_6 
2871365                    pepatac.py_ALL_7 
2871366                    pepatac.py_ALL_8 
2871367                   pepatac.py_ALL_10 
2871368                   pepatac.py_CML_17 
2871369                   pepatac.py_CML_18 
2871371                   pepatac.py_CML_20 
2871374                   pepatac.py_CRC_37 
2871375                  pepatac.py_CTCL_38 
2871376                    pepatac.py_GBM_3 
2871377                   pepatac.py_GBM_22 
2871378                   pepatac.py_GBM_30 
2871379                   pepatac.py_GIST_6 
2871380               pepatac.py_cervical_8 
2871381              pepatac.py_melanoma_22 
2871382         pepatac.py_pancreas_alpha_2 
2871383      pepatac.py_pancreatic_cancer_1 
2871384       pepatac.py_prostate_cancer_43 
2871385       pepatac.py_synovial_sarcoma_8 
2871386       pepatac.py_synovial_sarcoma_9 
4710403                 pepatac.py_tutorial 
4710403.bat+                          batch 
4710403.0                        pepatac.py 
4710480                    pepatac.py_gold1 
4710480.bat+                          batch 
4710480.0                        pepatac.py 
4710481                    pepatac.py_gold2 
4710481.bat+                          batch 
4710481.0                        pepatac.py 
4710482                    pepatac.py_gold3 
4710482.bat+                          batch 
4710482.0                        pepatac.py 
4710483                    pepatac.py_gold4 
4710483.bat+                          batch 
4710483.0                        pepatac.py 
4710484                    pepatac.py_gold5 
4710484.bat+                          batch 
4710484.0                        pepatac.py 
4710489                 pepatac.py_tutorial 
4710489.bat+                          batch 
4710489.0                        pepatac.py 
4710492                    pepatac.py_gold1 
4710492.0                        pepatac.py 
4710493                    pepatac.py_gold2 
4710493.bat+                          batch 
4710493.0                        pepatac.py 
4710494                    pepatac.py_gold3 
4710494.0                        pepatac.py 
4710495                    pepatac.py_gold4 
4710495.0                        pepatac.py 
4710496                    pepatac.py_gold5 
4710496.0                        pepatac.py 
4711284                    pepatac.py_gold2 
4711284.0                        pepatac.py 
[30m[[33m12:45 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msacct --format="JOBID, JOBNAME%30"[Kscancel 4710482
[30m[[33m12:45 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp  R    2:06:46      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:06:46      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:07:16      1 udc-ba25-12a
           4710494  standard pepatac.   jps3dp  R    2:07:16      1 udc-ba27-32a
[30m[[33m12:47 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp[1Pcancel 4710482[K[K94
[30m[[33m12:47 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4710494queue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710494  standard pepatac.   jps3dp CG    2:07:30      1 udc-ba27-32a
           4710495  standard pepatac.   jps3dp  R    2:07:01      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:07:01      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:07:31      1 udc-ba25-12a
[30m[[33m12:47 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp  R    2:07:33      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:07:33      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:08:03      1 udc-ba25-12a
[30m[[33m12:47 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mcd ../
[30m[[33m12:48 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mresults_pipeline(B[m[30m](B[m$ (B[mcd ../squeue -u jps3dp[1Pcancel 4710494queue -u jps3dp[1Pcancel 4710482acct --format="JOBID, JOBNAME%30"[18Pqueue -u jps3dptail -f PEPATAC_log.md [7Psqueue -u jps3dptail -f PEPATAC_log.md [24@looper run /scratch/jps3dp/DATA/gold_hg38.yaml[C
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Skipping sample 'gold2' for pipeline 'PEPATAC', flag found: PEPATAC_failed.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Job script (n=1; 4.70 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Submitted batch job 4711287
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Skipping sample 'gold5' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
4 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 1 of 5
Jobs submitted: 1
[0m[30m[[33m12:48 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mresults_pipeline(B[m[30m](B[m$ (B[mtail -f gold3/PEP[K[K[KPEP[K[K[K
fastq/                           PEPATAC_cleanup.sh               PEPATAC_log.md                   PEPATAC_running.flag             stats.tsv                        
lock.fastq__gold3_R1_trim.fastq  PEPATAC_commands.sh              PEPATAC_profile.tsv              raw/                             
[30m[[33m12:48 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mresults_pipeline(B[m[30m](B[m$ (B[mtail -f gold3/PEPATAC_log.md 
> `Fastq_reads`	166092252	PEPATAC	_RES_

### Adapter trimming:  (05-20 12:50:48) elapsed: 128.0 _TIME_

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold3/fastq/gold3_R1_trim.fastq`


> `skewer -f sanger -t 8 -m pe -x /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/NexteraPE-PE.fa --quiet -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold3/fastq/gold3 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold3/fastq/gold3_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold3/fastq/gold3_R2.fastq.gz` (299853)

<pre>
^C
[30m[[33m12:53 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mresults_pipeline(B[m[30m](B[m$ (B[mpigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | s ort -u
3' UTR
5' UTR
Exon
Intron
Promoter
Promoter Flanking Region
[30m[[33m12:53 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mresults_pipeline(B[m[30m](B[m$ (B[mcd ../[K[K[K[K[K[Kpwd
/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline
[30m[[33m12:55 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mresults_pipeline(B[m[30m](B[m$ (B[mcd gold2/
[30m[[33m12:56 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mpythong[K
Python 2.7.15 |Anaconda, Inc.| (default, Oct 10 2018, 21:32:13) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from argparse import ArgumentParser
>>> import os
>>> import re
>>> import sys
>>> import tempfile
>>> import tarfile
>>> import pypiper

>>> from pypiper import build_command
>>> outfolder = ""[K/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/'[K"
>>> import subprocess
>>> output = subprocess.run("/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[K[Kpigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'run'
>>> output = subprocess.run("pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u")])[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[1@[
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'run'
>>> output = subprocess.run(["pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u"])[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[1P[1P[1P[1@c[1@a[1@l[1@l
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 172, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
>>> output = subprocess.call(["pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u"])[1P[1@1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 172, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
>>> output = subprocess.call(["pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold1/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u"])[1P[1@2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 172, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
>>> output = subprocess.call(["pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u"])[C[1P[1P[1P[1@a[1@n[1@n[1@o[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[1P[C[1@/[1@s[1@c[1@r[1@a[1@t[1@c[1@h[C[1@/[1@j[1@p[1@s[1@3[1@d[1@p[C[1@/[1@t[1@o[1@o[1@l[1@s[C[1@/[1@d[1@a[1@t[1@a[1@b[1@i[1@o[C[1@/[1@p[1@e[1@p[1@a[1@t[1@a[1@c
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 172, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
>>> exit()
[30m[[33m12:59 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mwhich python
/apps/software/standard/core/anaconda/5.2.0-py2.7/bin/python
[30m[[33m01:01 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mpython
Python 2.7.15 |Anaconda, Inc.| (default, Oct 10 2018, 21:32:13) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> s[KftList = "3' UTR
  File "<stdin>", line 1
    ftList = "3' UTR
                   ^
SyntaxError: EOL while scanning string literal
>>> 5' UTR
  File "<stdin>", line 1
    5' UTR
         ^
SyntaxError: EOL while scanning string literal
>>> Exon
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Exon' is not defined
>>> Intron
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Intron' is not defined
>>> Promoter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Promoter' is not defined
>>> Promoter Flanking Region
  File "<stdin>", line 1
    Promoter Flanking Region
                    ^
SyntaxError: invalid syntax
>>> Promoter Flanking Region[C[C[C[C[C[C[C[C[C[C[C[C[K[C[C[C[C[2PIntron[C[C[C[C[2PExon5' UTR[C[C[C[C[10@ftList = "3[C[C[C[C[C\n5' UTR\nExpm[K[Kon\nIntron\m{P[K[K[KnPromoter\nPromoter Flanking Region"
>>> ftList
"3' UTR\n5' UTR\nExon\nIntron\nPromoter\nPromoter Flanking Region"
>>> str.splitlines(ftList)
["3' UTR", "5' UTR", 'Exon', 'Intron', 'Promoter', 'Promoter Flanking Region']
>>> ftList.splitlines()
["3' UTR", "5' UTR", 'Exon', 'Intron', 'Promoter', 'Promoter Flanking Region']
>>> ftList.split()
["3'", 'UTR', "5'", 'UTR', 'Exon', 'Intron', 'Promoter', 'Promoter', 'Flanking', 'Region']
>>> exit()
[30m[[33m01:04 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mls
aligned_hg38	    fastq   objects.tsv        PEPATAC_cleanup.sh   PEPATAC_failed.flag  PEPATAC_profile.tsv  QC_hg38  stats.tsv
aligned_hg38_exact  fastqc  peak_calling_hg38  PEPATAC_commands.sh  PEPATAC_log.md	 prealignments	      raw
[30m[[33m01:04 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mrm PEPATAC_failed.flag 
[30m[[33m01:04 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mlooper run /scratch/jps3dp/DATA/gold_hg38.yaml 
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Job script (n=1; 1.57 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711398
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Skipping sample 'gold5' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
4 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 1 of 5
Jobs submitted: 1
[0m[30m[[33m01:04 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4711398  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710495  standard pepatac.   jps3dp  R    2:24:02      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:24:02      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:24:32      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      15:43      1 udc-aj40-4c0
[30m[[33m01:04 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail[K[K[K[Ksqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp  R    2:25:09      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:25:09      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:25:39      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      16:50      1 udc-aj40-4c0
           4711398  standard pepatac.   jps3dp  R       0:00      1 udc-ba26-25
[30m[[33m01:05 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp  R    2:25:17      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:25:17      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:25:47      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      16:58      1 udc-aj40-4c0
           4711398  standard pepatac.   jps3dp  R       0:08      1 udc-ba26-25
[30m[[33m01:05 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md 
> `samtools idxstats /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam | grep -we 'chrM' -we 'chrMT' -we 'M' -we 'MT' -we 'rCRSd'| cut -f 3`


> `Mitochondrial_reads`	801.0	PEPATAC	_RES_
Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam`


> `samtools idxstats /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam | cut -f 1 | grep -vwe 'chrM' -vwe 'chrMT' -vwe 'M' -vwe 'MT' -vwe 'rCRSd'| xargs samtools view -b -@ 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam` (158276,158277,158278,158279)

<pre>
</pre>
Command completed. Elapsed time: 0:00:23. Running peak memory: 0.038GB.
  PID: 158276;	Command: samtools;	Return code: 0;	Memory used: 0.0GB
  PID: 158278;	Command: grep;	Return code: 0;	Memory used: 0.0GB
  PID: 158277;	Command: cut;	Return code: 0;	Memory used: 0.0GB
  PID: 158279;	Command: xargs;	Return code: 0;	Memory used: 0.038GB

> `mv /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam` (158619)

<pre>
</pre>
Command completed. Elapsed time: 0:00:00. Running peak memory: 0.038GB.
  PID: 158619;	Command: mv;	Return code: 0;	Memory used: 0.0GB

> `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam` (158620)

<pre>
</pre>
Command completed. Elapsed time: 0:00:15. Running peak memory: 0.038GB.
  PID: 158620;	Command: samtools;	Return code: 0;	Memory used: 0.0GB

### Calculate NRF, PBC1, and PBC2 (05-20 13:06:11) elapsed: 38.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_bamQC.tsv`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/bamQC.py -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam -c 8 -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_bamQC.tsv`
Command ID incremented by: `1`. Current ID: `30`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_unmap.bam`
Skipped command: `samtools view -b -@ 8 -f 12  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_unmap.bam`
Command ID incremented by: `1`. Current ID: `31`


### Remove dupes, build bigwig and bedgraph files (05-20 13:06:11) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`
Skipped command: `samtools sort -n -@ 8 -T /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpxQZaf9 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam | samtools view -h - -@ 8 | samblaster -r 2> /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_dedup_metrics_log.txt | samtools view -b - -@ 8 | samtools sort - -@ 8 -T /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpxQZaf9 -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`
Command ID incremented by: `5`. Current ID: `36`

Skipped command: `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`
Command ID incremented by: `1`. Current ID: `37`


### Produce smoothed and nucleotide-resolution tracks (05-20 13:06:11) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp/flag_completed`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/bamSitesToWig.py -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -c /nm/t1/genomes/hg38/hg38.chromSizes -b /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/gold2_shift.bed -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/gold2_exact.bw -w /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_smooth.bw -m atac -p 5`
Command ID incremented by: `1`. Current ID: `38`

Skipped command: `touch /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp/flag_completed`
Command ID incremented by: `1`. Current ID: `39`


### Calculate TSS enrichment (05-20 13:06:11) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.txt`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/pyTssEnrichment.py -a /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -b /nm/t1/genomes/hg38/hg38_TSS.tsv -p ends -c 8 -e 2000 -u -v -s 4 -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.txt`
Command ID incremented by: `1`. Current ID: `40`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_widthhg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp/flag_completed`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/bamSitesToWig.py -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -c /nm/t1/genomes/hg38/hg38.chromSizes -b /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/gold2_shift.bed -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/gold2_exact.bw -w /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_smooth.bw -m atac -p 5`
Command ID incremented by: `1`. Current ID: `38`

Skipped command: `touch /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp/flag_completed`
Command ID incremented by: `1`. Current ID: `39`


### Calculate TSS enrichment (05-20 13:06:11) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.txt`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/pyTssEnrichment.py -a /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -b /nm/t1/genomes/hg38/hg38_TSS.tsv -p ends -c 8 -e 2000 -u -v -s 4 -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.txt`
Command ID incremented by: `1`. Current ID: `40`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.pdf`
Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/PEPATAC_TSSenrichmentPlot.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.txt pdf`
Command ID incremented by: `1`. Current ID: `41`


> `TSS_Score`	12.5685253248	PEPATAC	_RES_
> `TSS enrichment`	QC_hg38/gold2_TssEnrichment.pdf	TSS enrichment	QC_hg38/gold2_TssEnrichment.png	PEPATAC	_OBJ_

### Plot fragment distribution (05-20 13:06:11) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLenDistribution.pdf`
Skipped command: `perl /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/fragment_length_dist.pl /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLen.txt`
Command ID incremented by: `1`. Current ID: `42`

Skipped command: `sort -n  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLen.txt | uniq -c  > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragCount.txt`
Command ID incremented by: `2`. Current ID: `44`

Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/fragment_length_dist.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLen.txt /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragCount.txt /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLenDistribution.pdf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLenDistribution.txt`
Command ID incremented by: `1`. Current ID: `45`

> `Fragment distribution`	QC_hg38/gold2_fragLenDistribution.pdf	Fragment distribution	QC_hg38/gold2_fragLenDistribution.png	PEPATAC	_OBJ_

### Call peaks (05-20 13:06:11) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.narrowPeak`
Skipped command: `macs2 callpeak -t /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/gold2_shift.bed --outdir /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38 -n gold2 -g hs --shift -75 --extsize 150 --nomodel --call-summits --nolambda --keep-dup all -p 0.01`
Command ID incremented by: `1`. Current ID: `46`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak`
Skipped command: `awk -v OFS='	' {$2 = $2 + 75 - 250; $3 = $3 - 75 + 250; print}' /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.narrowPeak > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak`
Command ID incremented by: `1`. Current ID: `47`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak`
Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/PEPATAC_reducePeaks.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak`
Command ID incremented by: `1`. Current ID: `48`


### Calculate fraction of reads in peaks (FRiP) (05-20 13:06:11) elapsed: 0.0 _TIME_


> `samtools view -@ 4 -c -L /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`

('14945152', u'5137187')

> `FRiP`	0.343736015532	PEPATAC	_RES_

### # Produce bigBed formatted narrowPeak file (05-20 13:06:19) elapsed: 8.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed`
Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/narrowPeakToBigBed.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes bedToBigBed /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed`
Command ID incremented by: `1`. Current ID: `49`


### Calculate peak coverage (05-20 13:06:19) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed`
Skipped command: `cut -f 1-3 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed`
Command ID incremented by: `1`. Current ID: `50`

Skipped command: `samtools view -H /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam | grep 'SN:' | awk -F':' '{print $2,$3}' | awk -F' ' -v OFS='	' '{print $1,$3}' > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt`
Command ID incremented by: `4`. Current ID: `54`

Skipped command: `bedtools sort -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed -faidx /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed`
Command ID incremented by: `1`. Current ID: `55`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_coverage.bed`
Skipped command: `bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed -b /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -g /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_coverage.bed`
Command ID incremented by: `1`. Current ID: `56`


### Calculate read coverage (05-20 13:06:19) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz`
Skipped command: `ln -sf /sfs/lustre/scratch/jps3dp/tools/databio/pepatac/anno/hg38_annotations.bed.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz`
Command ID incremented by: `1`. Current ID: `57`


> `pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u`

Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.f
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.f
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.r
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.c
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.c
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.g
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.f
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.r
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.u
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.n
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.g
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.g
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.3
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.8
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.3
Target to produce: `/`,`s`,`f`,`s`,`/`,`l`,`u`,`s`,`t`,`r`,`e`,`/`,`a`,`l`,`l`,`o`,`c`,`a`,`t`,`i`,`o`,`n`,`s`,`/`,`s`,`h`,`e`,`f`,`f`,`l`,`a`,`b`,`/`,`p`,`r`,`o`,`c`,`e`,`s`,`s`,`e`,`d`,`/`,`g`,`o`,`l`,`d`,`/`,`p`,`e`,`p`,`a`,`t`,`a`,`c`,`/`,`h`,`g`,`3`,`8`,`/`,`f`,`i`,`x`,`e`,`d`,`_`,`w`,`i`,`d`,`t`,`h`,`_`,`p`,`e`,`a`,`k`,`s`,`/`,`r`,`e`,`s`,`u`,`l`,`t`,`s`,`_`,`p`,`i`,`p`,`e`,`l`,`i`,`n`,`e`,`/`,`g`,`o`,`l`,`d`,`2`,`/`,`Q`,`C`,`_`,`h`,`g`,`3`,`8`,`/`,`3`,`'`,` `,`U`,`T`,`R`


> `pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | awk -F'	' '{print>"/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/"$4}'` (159071,159072)

<pre>
</pre>
Command completed. Elapsed time: 0:00:01. Running peak memory: 0.038GB.
  PID: 159072;	Command: awk;	Return code: 0;	Memory used: 0.0GB
  PID: 159071;	Command: pigz;	Return code: 0;	Memory used: 0.0GBg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam.bai
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpxQZaf9
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp/flag_completed
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed

### Pipeline failed at:  (05-20 13:06:22) elapsed: 2.0 _TIME_

Total time: 0:00:51
Failure reason: Pipeline failure. See details above.

Changed status from running to failed.
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
^[[A^[[A^C
[30m[[33m01:11 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md [7Psqueue -u jps3dplooper run /scratch/jps3dp/DATA/gold_hg38.yaml [C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[24Prm PEPATAC_failed.flag[C[24@looper run /scratch/jps3dp/DATA/gold_hg38.yaml[C
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Job script (n=1; 1.57 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711403
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requestsssed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711403
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Skipping sample 'gold5' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
4 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 1 of 5
Jobs submitted: 1
[0m[30m[[33m01:11 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mlooper run /scratch/jps3dp/DATA/gold_hg38.yaml [C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[24Ptail -f PEPATAC_log.md[C
Exception: Pipeline failure. See details above.
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
^C
[30m[[33m01:17 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4711403  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710495  standard pepatac.   jps3dp  R    2:36:57      1 udc-ba27-32a
           4710496  standard pepatac.   jps3dp  R    2:36:57      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:37:27      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      28:38      1 udc-aj40-4c0
[30m[[33m01:17 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dptail -f PEPATAC_log.md 
Exception: Pipeline failure. See details above.
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
^C
[30m[[33m01:17 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msacct --format="JobID,JobName%30"
       JobID                        JobName 
------------ ------------------------------ 
2870870              pepatac.py_melanoma_22 
2870889      pepatac.py_pancreati481.bat+                          batch 
4710481.0                        pepatac.py 
4710482                    pepatac.py_gold3 
4710482.bat+                          batch 
4710482.0                        pepatac.py 
4710483                    pepatac.py_gold4 
4710483.bat+                          batch 
4710483.0                        pepatac.py 
4710484                    pepatac.py_gold5 
4710484.bat+                          batch 
4710484.0                        pepatac.py 
4710489                 pepatac.py_tutorial 
4710489.bat+                          batch 
4710489.0                        pepatac.py 
4710492                    pepatac.py_gold1 
4710492.0                        pepatac.py 
4710493                    pepatac.py_gold2 
4710493.bat+                          batch 
4710493.0                        pepatac.py 
4710494                    pepatac.py_gold3 
4710494.bat+                          batch 
4710494.0                        pepatac.py 
4710495                    pepatac.py_gold4 
4710495.0                        pepatac.py 
4710496                    pepatac.py_gold5 
4710496.0                        pepatac.py 
4711284                    pepatac.py_gold2 
4711284.bat+                          batch 
4711284.0                        pepatac.py 
4711287                    pepatac.py_gold3 
4711287.0                        pepatac.py 
4711398                    pepatac.py_gold2 
4711398.bat+                          batch 
4711398.0                        pepatac.py 
4711403                    pepatac.py_gold2 
[30m[[33m01:18 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4710495
[30m[[33m01:18 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp CG    2:37:51      1 udc-ba27-32a
           4711403  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710496  standard pepatac.   jps3dp  R    2:37:58      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:38:28      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      29:39      1 udc-aj40-4c0
[30m[[33m01:18 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp CG    2:37:51      1 udc-ba27-32a
           4711403  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710496  standard pepatac.   jps3dp  R    2:38:01      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:38:31      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      29:42      1 udc-aj40-4c0
[30m[[33m01:18 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710495  standard pepatac.   jps3dp CG    2:37:51      1 udc-ba27-32a
           4711403  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710496  standard pepatac.   jps3dp  R    2:38:15      1 udc-aj40-2c0
           4710492  standard pepatac.   jps3dp  R    2:38:45      1 udc-ba25-12a
           4711287  standard pepatac.   jps3dp  R      29:56      1 udc-aj40-4c0
[30m[[33m01:18 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4710492
[30m[[33m01:19 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4710492queue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4710492  standard pepatac.   jps3dp CG    2:39:18      1 udc-ba25-12a
           4711403  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4710496  standard pepatac.   jps3dp  R    2:38:50      1 udc-aj40-2c0
           4711287  standard pepatac.   jps3dp  R      30:31      1 udc-aj40-4c0
[30m[[33m01:19 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp[1Pcancel 4710492queue -u jps3dp[1Pcancel 4710495acct --format="JobID,JobName%30"[10Ptail -f PEPATAC_log.md [7Psqueue -u jps3dptail -f PEPATAC_log.md [24@looper run /scratch/jps3dp/DATA/gold_hg38.yaml[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[24Ptail -f PEPATAC_log.md[C[24@looper run /scratch/jps3dp/DATA/gold_hg38.yaml[C
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Job script (n=1; 6.34 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Submitted batch job 4711408
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Skipping sample 'gold2' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Job script (n=1; 3.06 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Submitted batch job 4711409
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Skipping sample 'gold5' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
3 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 2 of 5
Jobs submitted: 2
[0m[30m[[33m01:19 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md 

Skipped command: `bedtools sort -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed -faidx /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed`
Command ID incremented by: `1`. Current ID: `55`

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_coverage.bed`


> `bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed -b /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -g /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_coverage.bed` (95445)

<pre>
</pre>
Command completed. Elapsed time: 0:00:23. Running peak memory: 0.037GB.
  PID: 95445;	Command: bedtools;	Return code: 0;	Memory used: 0.01GB

### Calculate read coverage (05-20 13:20:27) elapsed: 23.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz`
Skipped command: `ln -sf /sfs/lustre/scratch/jps3dp/tools/databio/pepatac/anno/hg38_annotations.bed.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz`
Command ID incremented by: `1`. Current ID: `57`


> `pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u`

Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.f
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.f
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.r
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.c
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.c
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.g
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.f
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after te Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.a
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.r
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.u
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.t
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.p
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.i
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.n
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.e
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.g
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.o
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.l
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.d
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock._
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.h
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.g
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.3
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.8
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.__
Lock file created after test! Looping again: /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.3
Target to produce: `/`,`s`,`f`,`s`,`/`,`l`,`u`,`s`,`t`,`r`,`e`,`/`,`a`,`l`,`l`,`o`,`c`,`a`,`t`,`i`,`o`,`n`,`s`,`/`,`s`,`h`,`e`,`f`,`f`,`l`,`a`,`b`,`/`,`p`,`r`,`o`,`c`,`e`,`s`,`s`,`e`,`d`,`/`,`g`,`o`,`l`,`d`,`/`,`p`,`e`,`p`,`a`,`t`,`a`,`c`,`/`,`h`,`g`,`3`,`8`,`/`,`f`,`i`,`x`,`e`,`d`,`_`,`w`,`i`,`d`,`t`,`h`,`_`,`p`,`e`,`a`,`k`,`s`,`/`,`r`,`e`,`s`,`u`,`l`,`t`,`s`,`_`,`p`,`i`,`p`,`e`,`l`,`i`,`n`,`e`,`/`,`g`,`o`,`l`,`d`,`2`,`/`,`Q`,`C`,`_`,`h`,`g`,`3`,`8`,`/`,`3`,`'`,` `,`U`,`T`,`R`


> `pigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | awk -F'	' '{print>"/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/"$4}'` (97656,97657)

<pre>
</pre>
Command completed. Elapsed time: 0:00:01. Running peak memory: 0.037GB.
  PID: 97657;	Command: awk;	Return code: 0;	Memory used: 0.0GB
  PID: 97656;	Command: pigz;	Return code: 0;	Memory used: 0.0GB
Traceback (most recent call last):
  File "/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.py", line 1519, in <module>
    sys.exit(main())
  File "/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.py", line 1384, in main
    pm.run(cmd2, annoFile, container=pm.container)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 786, in run
    os.remove(lock_file)  # Remove lock file
OSError: [Errno 2] No such file or directory: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/lock.s'
Pipeline status: running
Starting cleanup: 14 files; 3 conditional files for cllts_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt
- /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed

### Pipeline failed at:  (05-20 13:20:29) elapsed: 2.0 _TIME_

Total time: 0:01:10
Failure reason: Pipeline failure. See details above.

Changed status from running to failed.
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
^C
[30m[[33m01:28 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mpigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u
3' UTR
5' UTR
Exon
Intron
Promoter
Promoter Flanking Region
[30m[[33m01:29 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mpigz -f -p 8 -d -c /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/hg38_annotations.bed.gz | cut -f 4 | sort -u[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[Klooper run /scratch/jps3dp/DATA/gold_hg38.yaml 
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Job script (n=1; 1.57 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711421
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Skipping sample 'gold5' for pipeline 'PEPATAC', flag found: PEPATAC_failed.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
4 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 1 of 5
Jobs submitted: 1
[0m[30m[[33m01:31 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4711421  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4711287  standard pepatac.   jps3dp  R      42:34      1 udc-aj40-4c0
           4711408  standard pepatac.   jps3dp  R      10:27      1 udc-ba25-12a
           4711409  standard pepatac.   jps3dp  R      10:27      1 udc-ba27-32a
[30m[[33m01:31 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md [7Psqueue -u jps3dplooper run /scratch/jps3dp/DATA/gold_hg38.yaml 
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Job script (n=1; 1.57 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711424
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
Job script (n=1; 3.20 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
Submitted batch job 4711425
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
3 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 2 of 5
Jobs submitted: 2
[0m[30m[[33m01:31 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mlooper run /scratch/jps3dp/DATA/gold_hg38.yaml [C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[31Psqueue -u jps3dplooper run /scratch/jps3dp/DATA/gold_hg38.yaml [C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[C[31Psqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4711421  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4711424  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4711425  standard pepatac.   jps3dp PD       0:00      1 (Priority)
           4711287  standard pepatac.   jps3dp  R      43:10      1 udc-aj40-4c0
           4711408  standard pepatac.   jps3dp  R      11:03      1 udc-ba25-12a
           4711409  standard pepatac.   jps3dp  R      11:03      1 udc-ba27-32a
[30m[[33m01:31 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4711421
[30m[[33m01:31 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4711421[K4
[30m[[33m01:31 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mscancel 4711424[K5
[30m[[33m01:32 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           4711287  standard pepatac.   jps3dp  R      43:25      1 udc-aj40-4c0
           4711408  standard pepatac.   jps3dp  R      11:18      1 udc-ba25-12a
           4711409  standard pepatac.   jps3dp  R      11:18      1 udc-ba27-32a
[30m[[33m01:32 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[msqueue -u jps3dp[1Pcancel 471142541queue -u jps3dplooper run /scratch/jps3dp/DATA/gold_hg38.yaml 
Command: run (Looper version: 0.12.0)
Loading divvy config file: /nm/t1/databio/divcfg/uva_rivanna.yaml
Activating compute package 'default'
Setting sample sheet from file '/sfs/lustre/scratch/jps3dp/tools/databio//pepatac_testing/metadata/gold_atac_annotation.csv'
Finding pipelines for protocol(s): ATAC-seq
Known protocols: ATAC, ATAC-SEQ, ATAC-seq
[36m## [1 of 5] gold1 (ATAC-seq)[0m
> Skipping sample 'gold1' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [2 of 5] gold2 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Job script (n=1; 1.57 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold2.sub
Submitted batch job 4711426
[36m## [3 of 5] gold3 (ATAC-seq)[0m
> Skipping sample 'gold3' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [4 of 5] gold4 (ATAC-seq)[0m
> Skipping sample 'gold4' for pipeline 'PEPATAC', flag found: PEPATAC_running.flag
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
[36m## [5 of 5] gold5 (ATAC-seq)[0m
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'motif' for option '--motif'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'TSS_name' for option '--TSS-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'anno_name' for option '--anno-name'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'extend' for option '--extend'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'lite' for option '--lite'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'deduplicator' for option '--deduplicator'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'blacklist' for option '--blacklist'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'peak_caller' for option '--peak-caller'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'no_fifo' for option '--noFIFO'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
Job script (n=1; 3.20 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
Submitted batch job 4711427
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
3 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 2 of 5
Jobs submitted: 2
[0m[30m[[33m01:32 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md 
Exception: Pipeline failure. See details above.
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", O'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'frip_ref_peaks' for option '--frip-ref-peaks'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'trimmer' for option '--trimmer'
> Note (missing optional attribute): 'pepatac.py' requests sample attribute 'keep' for option '--keep'
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
Job script (n=1; 3.20 Gb): /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold5.sub
Submitted batch job 4711427
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
Writing script to /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub
3 script(s) for skipped samples:
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold1.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold3.sub
/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/submission/pepatac.py_gold4.sub

Looper finished
Samples valid for job generation: 5 of 5
Successful samples: 5 of 5
Commands submitted: 2 of 5
Jobs submitted: 2
[0m[30m[[33m01:32 PM [36mjps3dp[31m@[36mudc-ba34-36 [1m[37mgold2(B[m[30m](B[m$ (B[mtail -f PEPATAC_log.md 
Exception: Pipeline failure. See details above.
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/apps/software/standard/core/anaconda/5.2.0-py2.7/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1744, in _exit_handler
    self.fail_pipeline(Exception("Pipeline failure. See details above."))
  File "/home/jps3dp/.local/lib/python2.7/site-packages/pypiper/manager.py", line 1638, in fail_pipeline
    raise e
Exception: Pipeline failure. See details above.
### [Pipeline run code and environment:]

*              Command:  `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.py --single-or-paired paired --genome hg38 --sample-name gold2 --input /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz --input2 /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz --prealignments rCRSd_3k human_repeats --peak-type fixed --genome-size hs -O /sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/results_pipeline -P 8 -M 16000`
*         Compute host:  udc-ba27-14
*          Working dir:  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2
*            Outfolder:  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/
*  Pipeline started at:   (05-20 13:34:58) elapsed: 1.0 _TIME_

### [Version log:]

*       Python version:  2.7.15
*          Pypiper dir:  `/sfs/qumulo/qhome/jps3dp/.local/lib/python2.7/site-packages/pypiper`
*      Pypiper version:  0.11.0
*         Pipeline dir:  `/sfs/lustre/scratch/jps3dp/tools/databio/pepatac/pipelines`
*     Pipeline version:  0.8.6
*        Pipeline hash:  a147ae41573786561ef0bfc35518da8210175b17
*      Pipeline branch:  * dev
*        Pipeline date:  2019-05-16 15:52:56 -0400
*        Pipeline diff:  3 files changed, 2042 insertions(+), 178 deletions(-)

### [Arguments passed to pipeline:]

*             `input2`:  `['/sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz']`
*         `paired_end`:  `True`
*              `motif`:  `False`
*               `keep`:  `False`
*      `prealignments`:  `['rCRSd_3k', 'human_repeats']`
*          `peak_type`:  `fixed`
*          `new_start`:  `False`
*            `recover`:  `False`
*     `frip_ref_peaks`:  `None`
*           `TSS_name`:  `None`
*       `deduplicator`:  `samblaster`
*        `sample_name`:  `gold2`
*               `lite`:  `False`
*              `input`:  `['/sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz']`
*        `genome_size`:  `hs`
*        `config_file`:  `pepatac.yaml`
*             `extend`:  `250`
*                `mem`:  `16000`
*       `force_follow`:  `False`
*          `blacklist`:  `None`
*              `cores`:  `8`
*          `anno_name`:  `None`
*            `trimmer`:  `skewer`
*        `peak_caller`:  `macs2`
*      `output_parent`:  `/sfs/lustre/allocations/shefflab/processed//gold/pepatac/hg38/fixed_width_peaks/results_pipeline`
*   `single_or_paired`:  `paired`
*              `dirty`:  `False`
*    `genome_assembly`:  `hg38`
*            `no_fifo`:  `False`

----------------------------------------


Changed status from initializing to running.

Loading config file: /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/pipelines/pepatac.yaml

Local input file: /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz
Local input file: /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz

> `File_mb`	1603.8181	PEPATAC	_RES_

> `Read_type`	paired	PEPATAC	_RES_

> `Genome`	hg38	PEPATAC	_RES_

### Merge/link and fastq conversion:  (05-20 13:34:58) elapsed: 1.0 _TIME_

Number of input file sets: 2
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz`
Command ID incremented by: `1`. Current ID: `1`

Local input file: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz'
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/data/sra_fastq//SRR5210450_2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `2`

Local input file: '/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz'
Found .fastq.gz file
Found .fq.gz file; no conversion necessary
Found .fastq.gz file
Found .fq.gz file; no conversion necessary
['ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz', 'ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz']
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz`
Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz`
Command ID incremented by: `1`. Current ID: `3`

Skipped command: `ln -sf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/raw/gold2_R2.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `4`


### Adapter trimming:  (05-20 13:34:58) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq`
Skipped command: `skewer -f sanger -t 8 -m pe -x /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/NexteraPE-PE.fa --quiet -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1.fastq.gz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2.fastq.gz`
Command ID incremented by: `1`. Current ID: `5`

Skipped command: `mv /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2-trimmed-pair1.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq`
Command ID incremented by: `1`. Current ID: `6`

Skipped command: `mv /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2-trimmed-pair2.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2_trim.fastq`
Command ID incremented by: `1`. Current ID: `7`


### Prealignments (05-20 13:34:58) elapsed: 0.0 _TIME_

Prealignment assemblies: ['rCRSd_3k', 'human_repeats']

### Map to rCRSd_3k (05-20 13:34:58) elapsed: 0.0 _TIME_

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2`


> `mkfifo /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2` (25921)

<pre>
</pre>
Command completed. Elapsed time: 0:00:00. Running peak memory: 0.0GB.
  PID: 25921;	Command: mkfifo;	Return code: 0;	Memory used: 0.0GB
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq.gz`
Skipped command: `perl /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R2_trim.fastq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `9`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq.gz`
Skipped command: `(bowtie2 -p 8 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nm/t1/genomes/rCRSd_3k/indexed_bowtie2/rCRSd_3k --rg-id gold2 -U /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/fastq/gold2_R1_trim.fastq --un /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/rCRSd_3k_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log`
Command ID incremented by: `1`. Current ID: `10`


> `grep 'aligned exactly 1 time' /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_bt_aln_summary.log | awk '{print $1}'`


> `Aligned_reads_rCRSd_3k`	11074130.0	PEPATAC	_RES_

> `Alignment_rate_rCRSd_3k`	38.98	PEPATAC	_RES_

### Map to human_repeats (05-20 13:34:59) elapsed: 0.0 _TIME_

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2`


> `mkfifo /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2` (25928)

<pre>
</pre>
Command completed. Elapsed time: 0:00:00. Running peak memory: 0.0GB.
  PID: 25928;	Command: mkfifo;	Return code: 0;	Memory used: 0.0GB
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_bt_aln_summary.log`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq.gz`
Skipped command: `perl /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `12`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_bt_aln_summary.log`
Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq.gz`
Skipped command: `(bowtie2 -p 8 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nm/t1/genomes/human_repeats/indexed_bowtie2/human_repeats --rg-id gold2 -U /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq --un /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/human_repeats_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_bt_aln_summary.log`
Command ID incremented by: `1`. Current ID: `13`


> `grep 'aligned exactly 1 time' /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_bt_aln_summary.log | awk '{print $1}'`


> `Aligned_reads_human_repeats`	1147860.0	PEPATAC	_RES_

> `Alignment_rate_human_repeats`	4.04	PEPATAC	_RES_

### Compress all unmapped read files (05-20 13:34:59) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R1.fq`
Command ID incremented by: `1`. Current ID: `14`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_rCRSd_3k_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `15`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq`
Command ID incremented by: `1`. Current ID: `16`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq.gz`
Skipped command: `pigz -f -p 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq`
Command ID incremented by: `1`. Current ID: `17`


### Map to genome (05-20 13:34:59) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam`
Skipped command: `bowtie2 -p 8  --very-sensitive -X 2000 --rg-id gold2 -x /nm/t1/genomes/hg38/indexed_bowtie2/hg38 -1 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R1.fq.gz -2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/prealignments/gold2_human_repeats_unmap_R2.fq.gz | samtools view -bS - -@ 1  | samtools sort - -@ 1 -T /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpUE31l4 -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam`
Command ID incremented by: `3`. Current ID: `20`

Skipped command: `samtools view -b -q 10 -@ 8 -U /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_fail_qc.bam -f 2 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam`
Command ID incremented by: `1`. Current ID: `21`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam.bai`
Skipped command: `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam`
Command ID incremented by: `1`. Current ID: `22`

Skipped command: `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam`
Command ID incremented by: `1`. Current ID: `23`


> `samtools idxstats /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam | grep -we 'chrM' -we 'chrMT' -we 'M' -we 'MT' -we 'rCRSd'| cut -f 3`


> `Mitochondrial_reads`	801.0	PEPATAC	_RES_
Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam`


> `samtools idxstats /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam | cut -f 1 | grep -vwe 'chrM' -vwe 'chrMT' -vwe 'M' -vwe 'MT' -vwe 'rCRSd'| xargs samtools view -b -@ 8 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam` (25937,25938,25939,25940)

<pre>
</pre>
Command completed. Elapsed time: 0:00:22. Running peak memory: 0.037GB.
  PID: 25937;	Command: samtools;	Return code: 0;	Memory used: 0.0GB
  PID: 25939;	Command: grep;	Return code: 0;	Memory used: 0.0GB
  PID: 25938;	Command: cut;	Return code: 0;	Memory used: 0.0GB
  PID: 25940;	Command: xargs;	Return code: 0;	Memory used: 0.037GB

> `mv /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_noMT.bam /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam` (26475)

<pre>
</pre>
Command completed. Elapsed time: 0:00:00. Running peak memory: 0.037GB.
  PID: 26475;	Command: mv;	Return code: 0;	Memory used: 0.0GB

> `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam` (26476)

<pre>
</pre>
Command completed. Elapsed time: 0:00:14. Running peak memory: 0.037GB.
  PID: 26476;	Command: samtools;	Return code: 0;	Memory used: 0.0GB

### Calculate NRF, PBC1, and PBC2 (05-20 13:35:35) elapsed: 37.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_bamQC.tsv`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/bamQC.py -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam -c 8 -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_bamQC.tsv`
Command ID incremented by: `1`. Current ID: `30`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_unmap.bam`
Skipped command: `samtools view -b -@ 8 -f 12  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_temp.bam > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_unmap.bam`
Command ID incremented by: `1`. Current ID: `31`


### Remove dupes, build bigwig and bedgraph files (05-20 13:35:35) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`
Skipped command: `samtools sort -n -@ 8 -T /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpToa8iz /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort.bam | samtools view -h - -@ 8 | samblaster -r 2> /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_dedup_metrics_log.txt | samtools view -b - -@ 8 | samtools sort - -@ 8 -T /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/tmpToa8iz -o /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`
Command ID incremented by: `5`. Current ID: `36`

Skipped command: `samtools index /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`
Command ID incremented by: `1`. Current ID: `37`


### Produce smoothed and nucleotide-resolution tracks (05-20 13:35:35) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/temp/flag_completed`
Skipped command: `/sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/bamSitesToWig.py -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_wids/results_pipeline/gold2/QC_hg38/gold2_TssEnrichment.txt pdf`
Command ID incremented by: `1`. Current ID: `41`


> `TSS_Score`	12.5685253248	PEPATAC	_RES_
> `TSS enrichment`	QC_hg38/gold2_TssEnrichment.pdf	TSS enrichment	QC_hg38/gold2_TssEnrichment.png	PEPATAC	_OBJ_

### Plot fragment distribution (05-20 13:35:35) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLenDistribution.pdf`
Skipped command: `perl /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/fragment_length_dist.pl /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLen.txt`
Command ID incremented by: `1`. Current ID: `42`

Skipped command: `sort -n  /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLen.txt | uniq -c  > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragCount.txt`
Command ID incremented by: `2`. Current ID: `44`

Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/fragment_length_dist.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLen.txt /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragCount.txt /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLenDistribution.pdf /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/QC_hg38/gold2_fragLenDistribution.txt`
Command ID incremented by: `1`. Current ID: `45`

> `Fragment distribution`	QC_hg38/gold2_fragLenDistribution.pdf	Fragment distribution	QC_hg38/gold2_fragLenDistribution.png	PEPATAC	_OBJ_

### Call peaks (05-20 13:35:35) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.narrowPeak`
Skipped command: `macs2 callpeak -t /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38_exact/gold2_shift.bed --outdir /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38 -n gold2 -g hs --shift -75 --extsize 150 --nomodel --call-summits --nolambda --keep-dup all -p 0.01`
Command ID incremented by: `1`. Current ID: `46`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak`
Skipped command: `awk -v OFS='	' {$2 = $2 + 75 - 250; $3 = $3 - 75 + 250; print}' /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.narrowPeak > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak`
Command ID incremented by: `1`. Current ID: `47`

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak`
Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/PEPATAC_reducePeaks.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak`
Command ID incremented by: `1`. Current ID: `48`


### Calculate fraction of reads in peaks (FRiP) (05-20 13:35:35) elapsed: 0.0 _TIME_


> `samtools view -@ 4 -c -L /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam`

('14945152', u'5137187')

> `FRiP`	0.343736015532	PEPATAC	_RES_

### # Produce bigBed formatted narrowPeak file (05-20 13:35:43) elapsed: 7.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed`
Skipped command: `Rscript /sfs/lustre/scratch/jps3dp/tools/databio//pepatac/tools/narrowPeakToBigBed.R /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak /nm/t1/genomes/hg38/hg38.chromSizes bedToBigBed /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bigBed`
Command ID incremented by: `1`. Current ID: `49`


### Calculate peak coverage (05-20 13:35:43) elapsed: 0.0 _TIME_

Target exists: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed`
Skipped command: `cut -f 1-3 /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_fixedWidth_normalized.narrowPeak > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed`
Command ID incremented by: `1`. Current ID: `50`

Skipped command: `samtools view -H /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam | grep 'SN:' | awk -F':' '{print $2,$3}' | awk -F' ' -v OFS='	' '{print $1,$3}' > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt`
Command ID incremented by: `4`. Current ID: `54`

Skipped command: `bedtools sort -i /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks.bed -faidx /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed`
Command ID incremented by: `1`. Current ID: `55`

Target to produce: `/sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_coverage.bed`


> `bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_sort.bed -b /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/aligned_hg38/gold2_sort_dedup.bam -g /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/gold/pepatac/hg38/fixed_width_peaks/results_pipeline/gold2/peak_calling_hg38/gold2_peaks_coverage.bed` (26826)

<pre>
</pre>
Command completed. Elapsed time: 0:00:25. Running peak memory: 0.037GB.
  PID: 26826;	Command: bedtools;	Return code: 0;	Memory used: 0.009GB

### Calculate read coverage (05-20 13:36:07) elapsed: 25.0 _TIME_

Skipping read annotation...
This requires a hg38 annotation file.
Confirm this file is present in /sfs/lustre/scratch/jps3dp/tools/databio/pepatac/anno or specify using `--anno-name`

### Plot FRiP/F (05-20 13:36:07) elapsed: 0.0 _TIME_


> `samtools view -@ 8 -q 10